docs/geps/gep-03.rst

```diff
@@ -0,0 +1,296 @@
+            en: >
+                Shares of income which do not lead to tapering of Arbeitslosengeld II
+                benefits.
+
+3. The YYYY-MM-DD key(s)
+
+   * hold all historical values for a specific parameter or set of parameters in the
+     ``value`` subkey;
+   * is present with ``value: null`` if a parameter ceases to exist starting on a
+     particular date;
+   * contain a precise reference to the law in the ``reference`` subkey;
+   * may add additional descriptions in the ``note`` key;
+   * may give hints towards the type of function they refer to via the ``type`` subkey;
+   * may include formulas if the law does;
+   * may reference other parameters as described below.
+
+   The remainder of this section explains this element in much more detail.
+
+
+[YYYY-MM-DD].reference
+++++++++++++++++++++++
+
+* cites the law (Gesetz "G. v."), decree (Verordnung "V. v.") or proclamation
+  (Bekanntmachung "B. v.") that changes the law
+* in German
+* follows the style ``Artikel [n] [G./V./B.] v. [DD.MM.YYYY] BGBl. I S. [SSSS].``
+* do not add information "geändert durch" (it is always a change) or the date the law
+  comes into force (this is exactly the date key one level above)
+* the page should be the first page of the law/decree/proclamation, not the exact page
+  of the parameter
+
+Example:
+
+.. code-block:: yaml
+
+    reference: G. v. 24.12.2003 BGBl. I S. 2964.
+
+
+[YYYY-MM-DD].value
+++++++++++++++++++
+
+The general idea is to make the replication of the laws very obvious. If the law
+includes a table, we will have a dictionary with keys 0, 1, 2, .... If the law includes
+a formula, it should be included. Etc.
+
+The following walks through several cases.
+
+.. todo::
+
+    Make these cases close to exhaustive.
+
+* The simplest case is a single parameter, which should be specified as:
+
+  .. code-block:: yaml
+
+      value: 520
+
+* Values may reference another parameter if that is fixed by law in a given year. That
+  is, it must not depend on anything else or individual-level data. So this would not
+  work, for example, for income tax schedules. Example:
+
+  .. code-block:: yaml
+
+      value: arbeitsl_geld_2.regelbedarf * 0.6
+
+
+* More complex: A piecewise linear function
+
+    .. code-block:: yaml
+
+        type: piecewise_linear
+        value:
+            0:
+                lower_threshold: 0
+                upper_threshold: 400
+                rate: 0.15
+                intercept_at_lower_threshold: 0
+            1:
+                upper_threshold: 800
+                rate: 0.3
+            2:
+                upper_threshold: 1200
+                rate: 0.15
+            3:
+                upper_threshold: inf
+                rate: 0
+                note: Making this explicit.
+
+* Formulas: We want to allow for something like this if the law is specified that way
+  (e.g. the Steuertarif, IIRC):
+
+  .. code-block:: yaml
+
+      value: (2000 + 0.5 * x) * x
```

Comment:
not sure if this is such a good idea because it makes it more opaque where to look for a specific formula. Kind of an in-between case between parameter and policy structure. I see a use case though for soc sec contributions, where we use half the value in the law.

Reply:
Sorry, this was a bit too terse. and in the wrong order. See the annotation below: 

    Will require that a function is specified in the ``type`` keyword so that we can get standard
    representations via SymPy or the like.

So this is just an alternative way to specify parameters for a formula that exists among our generic functions. E.g., we would not need to expand the example formula to get the coefficients of `a + b * x + c * x **2`. We would not use the formula directly, so the code stays the same.